### PR TITLE
Update application state to include action default params

### PR DESCRIPTION
**Pull request by duynguyen**
Many users want to define default params for their actions. It is not yet explicitly mentioned in our docs about how to do it.
IMO, default params can be treated as (static) application state, hence creating this PR.

### CODE REVIEW

File: guides/application_state.md [hunk 1]
```diff
@@ -1,6 +1,44 @@
...
+hello-world:
+  function: actions/hello/index.js
+  runtime: 'nodejs:12'
+  inputs:
+    name: $NAME
+```
+
+### Considerations about security
+
+Many developers use the default params as a mechanism to provision actions with the secrets/passwords needed to authenticate against some other systems (databases, services, APIs).
```
Comment (meryllblanchet) [inline guides/application_state.md:35]: This paragraph could be misleading in the sense that the mechanism could be perceived as a workaround to manage application secrets at action level, while it is in fact the out-of-the-box solution for secret management within Runtime (and hence the recommended approach).

**Review by meryllblanchet — CHANGES_REQUESTED** [review]
Thanks for the contribution @duynguyen ! Please see my comment about secret management and think about how we could rephrase this section. Otherwise everything looks good!

File: guides/application_state.md (hunk 1 above)
Comment (duynguyen) [reply guides/application_state.md]: good point, I have updated the text to differentiate Adobe auth vs. third party auth.
Is it clearer?

**Review by meryllblanchet — APPROVED** [review]
LGTM

**Pull request merged**